middleware/static/static_test.go

```diff
@@ -0,0 +1,686 @@
+	require.NoError(t, err, "app.Test(req)")
+	require.Equal(t, 200, resp.StatusCode, "Status code")
+
+	resp, err = app.Test(httptest.NewRequest(fiber.MethodGet, "/static/", nil))
+	require.NoError(t, err, "app.Test(req)")
+	require.Equal(t, 200, resp.StatusCode, "Status code")
+
+	resp, err = app.Test(httptest.NewRequest(fiber.MethodGet, "/static/style.css", nil))
+	require.NoError(t, err, "app.Test(req)")
+	require.Equal(t, 200, resp.StatusCode, "Status code")
+
+	body, err = io.ReadAll(resp.Body)
+	require.NoError(t, err, "app.Test(req)")
+	require.Contains(t, string(body), "color")
+
+	app = fiber.New()
+	app.Get("/static*", New(dir))
+
+	resp, err = app.Test(httptest.NewRequest(fiber.MethodGet, "/static", nil))
+	require.NoError(t, err, "app.Test(req)")
+	require.Equal(t, 404, resp.StatusCode, "Status code")
+
+	resp, err = app.Test(httptest.NewRequest(fiber.MethodGet, "/static/", nil))
+	require.NoError(t, err, "app.Test(req)")
+	require.Equal(t, 404, resp.StatusCode, "Status code")
+
+	resp, err = app.Test(httptest.NewRequest(fiber.MethodGet, "/static/style.css", nil))
+	require.NoError(t, err, "app.Test(req)")
+	require.Equal(t, 200, resp.StatusCode, "Status code")
+
+	body, err = io.ReadAll(resp.Body)
+	require.NoError(t, err, "app.Test(req)")
+	require.Contains(t, string(body), "color")
+
+	app = fiber.New()
+	app.Get("/static*", New(dir))
+
+	resp, err = app.Test(httptest.NewRequest(fiber.MethodGet, "/static", nil))
+	require.NoError(t, err, "app.Test(req)")
+	require.Equal(t, 404, resp.StatusCode, "Status code")
+
+	resp, err = app.Test(httptest.NewRequest(fiber.MethodGet, "/static/", nil))
+	require.NoError(t, err, "app.Test(req)")
+	require.Equal(t, 404, resp.StatusCode, "Status code")
+
+	resp, err = app.Test(httptest.NewRequest(fiber.MethodGet, "/static/style.css", nil))
+	require.NoError(t, err, "app.Test(req)")
+	require.Equal(t, 200, resp.StatusCode, "Status code")
+
+	body, err = io.ReadAll(resp.Body)
+	require.NoError(t, err, "app.Test(req)")
+	require.Contains(t, string(body), "color")
+}
+
+func Test_Static_FS(t *testing.T) {
+	t.Parallel()
+
+	app := fiber.New()
+	app.Get("/*", New("", Config{
+		FS:     os.DirFS("../../.github/testdata/fs"),
+		Browse: true,
+	}))
+
+	resp, err := app.Test(httptest.NewRequest(fiber.MethodGet, "/", nil))
+	require.NoError(t, err, "app.Test(req)")
+	require.Equal(t, 200, resp.StatusCode, "Status code")
+	require.Equal(t, fiber.MIMETextHTMLCharsetUTF8, resp.Header.Get(fiber.HeaderContentType))
+
+	resp, err = app.Test(httptest.NewRequest(fiber.MethodGet, "/css/style.css", nil))
+	require.NoError(t, err, "app.Test(req)")
+	require.Equal(t, 200, resp.StatusCode, "Status code")
+	require.Equal(t, fiber.MIMETextCSSCharsetUTF8, resp.Header.Get(fiber.HeaderContentType))
+
+	body, err := io.ReadAll(resp.Body)
+	require.NoError(t, err, "app.Test(req)")
+	require.Contains(t, string(body), "color")
+}
+
+/*func Test_Static_FS_DifferentRoot(t *testing.T) {
```

Comment:
don´t forget this test -> its because of the needed fasthttp update or ?

Reply:
I commented it because of i am not sure it's the wanted behavior. If you use fs root has no functionality with current fasthttp fs implementation